src/Data/OpenUnion.hs

```diff
@@ -97,8 +97,8 @@ newtype P t r = P{unP :: Int}
 -- | Typeclass that asserts that effect @t@ is contained inside the effect-list
 -- @r@.
 --
--- The @FindElem@ typeclass is necessary for implementation reasons and is not
--- required for using the effect list.
+-- The @FindElem@ typeclass is a implementation detail and not required for
```

Comment:
@power-fungus typo: `s/a implementation/an implementation/`

some additional context, which you may or may not want to include. specifically `FindElem` is one way to implement some of the type-level set operations we need, specifically `Member.inj` and `Member.prj`. there are alternate implementations of `OpenUnion.Member` with different tradeoffs. `FindElem` is fairly good in terms of not requiring additional type level constraints (e.g., `Typeable`) which restrict use, while still being fairly performant. some other examples of alternate implementations can be found on Oleg's site

Reply:
@suhailshergill I would like to remove the context and just note that the `FindElem` Constraint should not be of interest to the user at this point. 

Maybe an advanced piece of documentation could point at the details and differences between different `Member` or `OpenUnion` implementations - a blog post maybe?